pkg/apis/psmdb/v1/psmdb_defaults.go

```diff
@@ -175,7 +175,27 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 				cr.Spec.Sharding.Mongos.Size = minSafeMongosSize
 			}
 		}
+		if cr.CompareVersion("1.15.0") >= 0 {
+			var fsgroup *int64
+			if platform == version.PlatformKubernetes {
+				var tp int64 = 1001
+				fsgroup = &tp
+			}
+
+			if cr.Spec.Sharding.Mongos.ContainerSecurityContext == nil {
+				tvar := true
```

Comment:
I don't think we should do this here by modifying user CR. I think we should do it directly where we define mongos deployment, in `pkg/psmdb/mongos.go`.

Wdyt @egegunes @hors

Reply:
Why is it better to do it in sts? @inelpandzic

Reply:
I don't think we should touch CR if not necessary, it can be easily removed by the user. In the end, it is something internal that we need to ensure, just like we, for example, set init container or sidecar containers. We don't do it by updating the CR, but internally set it in our sts/deployment specs.

Reply:
Aha, I just saw that we do the same thing for other objects, like backup, and in this case, I think we should do it as you did it now @nmarukovich because of the consistency even though I think that generally it should be done where we create specs themselves rather than in CR.

Reply:
> Aha, I just saw that we do the same thing for other objects, like backup, and in this case, I think we should do it as you did it now @nmarukovich because of the consistency even though I think that generally it should be done where we create specs themselves rather than in CR.

Despite the fact, that we already did it, I still want to discuss what we want to have ideally. In this case we can plan to refactor it.

Reply:
Ideally these should be (or close to) where we define sts for given component. Currently this place is the correct one IMO.

Reply:
Ok! So let's merge in this case (=